src/shared/vscodeRemoteSshConfig.ts

```diff
@@ -0,0 +1,229 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as fs from 'fs-extra'
+import * as nls from 'vscode-nls'
+import { getLogger } from './logger'
+import { ChildProcess, ChildProcessResult } from './utilities/childProcess'
+import { Result } from './utilities/result'
+import { ToolkitError } from './errors'
+import { getIdeProperties } from './extensionUtilities'
+import { showConfirmationMessage } from './utilities/messages'
+import { CancellationError } from './utilities/timeoutUtils'
+import { getSshConfigPath } from './extensions/ssh'
+import globals from './extensionGlobals'
+import { fileExists, readFileAsString } from './filesystemUtilities'
+
+const localize = nls.loadMessageBundle()
+
+export class VscodeRemoteSshConfig {
```

Comment:
Not really vscode-specific, just normal ssh stuff

```suggestion
export class SshConfig {
```